roadmap/implementers-guide/src/node/validity/assignments.md

```diff
@@ -0,0 +1,101 @@
+
+## Announcements / Notices
+
+We gossip assignment notices among nodes so that all validators know which validators should check each candidate, and if any candidate requires more checkers.  
+
+Assignment notices consist of a relay chain context given by a block hash, an assignment criteria, consisting of the criteria identifier and optionally a criteria specific field, an assignee identifier, and a VRF signature by the assignee, which itself consists of a VRF pre-output and a DLEQ proof.  Its VRF input consists of the criteria, usually including a criteria specific field, and a "story" about its relay chain context block. 
+
+We never include stories inside the gossip messages containing assignment notices, but require each validator reconstruct them.  We never care about assignments in the disputes process, so this does not complicate remote disputes.
+
+In a Schnorr VRF, there is an extra signed message distinct from this input, which we set to the relay chain block hash.  As a result, assignment notices are self signing and can be "politely" gossiped without additional signatures, meaning between nodes who can compute the story from the relay chain context.  In other words, if we cannot compute the story required by an assignment notice's VRF part then our self signing property fails and we cannot verify its origin.  We could fix this with either another signature layer (64 bytes) or by including the VRF input point computed from the story (32 bytes), but doing so appears unhelpful.
+
+Any validator could send their assignment notices and/or approval votes too early.  We gossip the approval votes early because they represent a major commitment by the validator.  We delay gossiping the assignment notices until they agree with our local clock however.  We also impose a politness condition that the recipient knows the relay chain context used by the assignment notice.
+
+## Stories
+
+We based assignment criteria upon two possible "stories" about the relay chain block `R` that declared the candidate available.  All stories have an output that attempts to minimize adversarial influence, which then acts as the VRF input for an assignment criteria.
+
+We first have a `RelayVRFStory` that outputs the randomness from another VRF output known only to the relay chain block producer who created `R`, and unknown even to them two epochs previously.  
+
+In BABE, we create this value calling `schnorrkel::vrf::VRFInOut::make_bytes` with a context "A&V RC-VRF", with the `VRFInOut` coming from either the VRF that authorized block production for primary blocks, or else form the secondary block VRF for the secondary block type.
```

Comment:
This is a piece of data that is then submitted to the chain? The writing here is unclear; it might also be interpreted as this story (and thus assignments based on it) is only instantiable by the block author of the context of the story, who must persist this private VRF output for some time.

Reply:
As a consensus process, we file incoming notices inside a data structure that contains the story.  As an on-chain check, yes I suppose each block creates a similar data structure inside the state that contains the story, along with any notices received. 

I've avoided thinking too much about the on-chain data structure because I'm worried first about the security aspects of the consensus process, but..  I've been considering some refactoring which as a by product permits doing the on-chain work more efficiently, so doing that tonight I guess..

Reply:
I see; it makes sense to mirror it in the chain state, at least up to some amount of time.

Can you clarify though, that my interpretation is correct and this story can be used by any validator and not just the author of the block? The language of the post here is confusing as it says "from another VRF output known only to the relay-chain block producer". So the VRF output is private, but the randomness is public - right?

Reply:
oh, I see you answered it below